test/test_local_io.py

```diff
@@ -721,6 +723,63 @@ def test_rar_archive_loader(self):
         data_refs = list(rar_loader_dp)
         self._unordered_compressed_files_comparison_helper(self.temp_files, data_refs)
 
+    def _add_data_to_wds_tar(self, archive, name, value):
+        if isinstance(value, str):
+            value = value.encode()
+        info = tarfile.TarInfo(name)
+        info.size = len(value)
+        archive.addfile(info, io.BytesIO(value))
+
+    def _create_wds_tar(self, dest):
+        with tarfile.open(dest, mode="w") as archive:
+            for i in range(10):
+                self._add_data_to_wds_tar(archive, f"data/{i}.txt", f"text{i}")
+                self._add_data_to_wds_tar(archive, f"data/{i}.bin", f"bin{i}")
+
+    def test_webdataset(self) -> None:
+        # Setup
+        self._create_wds_tar(os.path.join(self.temp_dir.name, "wds.tar"))
+
+        # Functional Test: groups samples correctly
+        source_dp = IterableWrapper(
+            # simulated tar file content
+            [
+                ("/path/to/file1.jpg", b"1"),
+                ("/path/to/_something_", b"nothing"),
+                ("/path/to/file1.cls", b"2"),
+                ("/path/to/file2.jpg", b"3"),
+                ("/path/to/file2.cls", b"4"),
+            ]
+        )
+        web_dataset = WebDataset(source_dp)
+        self.assertEqual(
+            # expected grouped output
+            [
+                {".jpg": b"1", ".cls": b"2", "__key__": "/path/to/file1"},
+                {".jpg": b"3", ".cls": b"4", "__key__": "/path/to/file2"},
+            ],
+            list(web_dataset),
+        )
+
+    def test_webdataset2(self) -> None:
+        # Setup
+        self._create_wds_tar(os.path.join(self.temp_dir.name, "wds.tar"))
```

Comment:
Maybe we should define `record_count = 10` and then pass it here?
```suggestion
        record_count = 10
        self._create_wds_tar(os.path.join(self.temp_dir.name, "wds.tar"), record_count)
```

It's not obvious that at the end of the test you are expecting `10` items.

Reply:
OK, added. I also am using it in the assertion later.